src/libraries/System.Threading.Tasks.Dataflow/tests/Dataflow/TransformManyBlockTests.IAsyncEnumerable.cs

```diff
@@ -0,0 +1,663 @@
+            {
+                barrier1.SignalAndWait();
+                // will test InputCount here
+                barrier2.SignalAndWait();
+                return DataflowTestHelpers.ToAsyncEnumerable(item);
+            }
+
+            TransformManyBlock<int, int> tb = new TransformManyBlock<int, int>(body);
+
+            for (int iter = 0; iter < 2; iter++)
+            {
+                tb.PostItems(1, 2);
+                for (int i = 1; i >= 0; i--)
+                {
+                    barrier1.SignalAndWait();
+                    Assert.Equal(expected: i, actual: tb.InputCount);
+                    barrier2.SignalAndWait();
+                }
+            }
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
+        [OuterLoop] // spins waiting for a condition to be true, though it should happen very quickly
+        public async Task TestCountAsyncEnumerable()
+        {
+            var tb = new TransformManyBlock<int, int>(DataflowTestHelpers.ToAsyncEnumerable);
+            Assert.Equal(expected: 0, actual: tb.InputCount);
+            Assert.Equal(expected: 0, actual: tb.OutputCount);
+
+            tb.PostRange(1, 11);
+            await Task.Run(() => SpinWait.SpinUntil(() => tb.OutputCount == 10));
+            for (int i = 10; i > 0; i--)
+            {
+                Assert.True(tb.TryReceive(out int item));
+                Assert.Equal(expected: 11 - i, actual: item);
+                Assert.Equal(expected: i - 1, actual: tb.OutputCount);
+            }
+        }
+
+        [Fact]
+        public async Task TestChainedSendReceiveAsyncEnumerable()
+        {
+            foreach (bool post in DataflowTestHelpers.BooleanValues)
+            {
+                static TransformManyBlock<int, int> func() => new TransformManyBlock<int, int>(i => AsyncEnumerable.Repeat(i * 2, 1));
+                var network = DataflowTestHelpers.Chain<TransformManyBlock<int, int>, int>(4, func);
+
+                const int Iters = 10;
+                for (int i = 0; i < Iters; i++)
+                {
+                    if (post)
+                    {
+                        network.Post(i);
+                    }
+                    else
+                    {
+                        await network.SendAsync(i);
+                    }
+                    Assert.Equal(expected: i * 16, actual: await network.ReceiveAsync());
+                }
+            }
+        }
+
+        [Fact]
+        public async Task TestSendAllThenReceiveAsyncEnumerable()
+        {
+            foreach (bool post in DataflowTestHelpers.BooleanValues)
+            {
+                static TransformManyBlock<int, int> func() => new TransformManyBlock<int, int>(i => AsyncEnumerable.Repeat(i * 2, 1));
+                var network = DataflowTestHelpers.Chain<TransformManyBlock<int, int>, int>(4, func);
+
+                const int Iters = 10;
```

Comment:
```suggestion
                const int iters = 10;
```
private constants should be written in lower case